streamingtest: TestStreamingReplanOnLag is flaky

```
panic: test timed out after 7m12s
```

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/13030958?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

The [test history](https://teamcity.cockroachdb.com/test/7443067493072918674?currentProjectId=Cockroach_Ci_Tests&expandTestHistoryChartSection=true) shows that it failed 39 out of the last 159 runs.

Jira issue: CRDB-34251

--- a/pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go
+++ b/pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go
@@ -758,6 +758,8 @@ func TestStreamingReplanOnLag(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
+	skip.UnderDuressWithIssue(t, 115850, "time to scatter ranges takes too long under duress")
+
 	ctx := context.Background()
 	args := replicationtestutils.DefaultTenantStreamingClustersArgs
 	args.MultitenantSingleClusterNumNodes = 3

